src/v/redpanda/application.h

```diff
@@ -256,6 +262,48 @@ class application {
 
     bool datalake_enabled();
 
+    // Stop the service.
+    // The method should be invoked in the ss::thread context.
+    template<class Service>
+    void stop_service(
+      Service& s,
+      ss::sstring name,
```

Comment:
tbh could probably just log something like `typeid(Service)` here, or add a `name()` method to all the services

Reply:
plus one, there is prior are for this in the state machines in raft, which also need a name

Reply:
did that